### PR TITLE
feat(experimental): add GPU BVH queries

### PR DESCRIPTION
**Pull request by ibgreen-openai**
## Overview

Adds exact point-containment and axis-aligned bounds-intersection traversal for the experimental flat GPU BVH.

## Concepts and use cases

- Traverses the complete-binary hierarchy one level at a time and prunes disjoint subtrees without a hidden traversal stack.
- Publishes stable IDs through capacity-bounded output, full match count, overflow, and optional source-ID mask contracts.
- Reports visited-node count so source ordering and future topology builders can be compared by useful traversal work.
- Supports viewport and box selection, object-level visibility, simulation neighborhoods, and picking broad phases while documenting when scan or grid paths are preferable.

## Changes

- Add `GPUBVHQuery` with mutable packed 2D/3D point and bounds queries.
- Add exact CPU-oracle tests for pruning, overlap, mutable queries, invalid input, masks, and overflow propagation.
- Add an Overview/Concepts documentation page, navigation, and roadmap status update.

## Verification

- `nvm use`
- `yarn lint fix`
- `yarn build`
- `yarn test-headless modules/experimental/test/gpu-primitives/gpu-bvh-query.spec.ts`
- Full `yarn test`: node suite passes and the command-graph texture CI regression passes; the run retains an unrelated `modules/arrow-layers/test/layers/arrow-layers.spec.ts` failure in files untouched by this stack.
- `(cd website && yarn build)`

## Stack

Base: #2815

### CODE REVIEW

File: modules/experimental/src/gpu-primitives/gpu-bvh-query.ts [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+  @builtin(global_invocation_id) globalId: vec3<u32>
+) {
+  if (globalId.x >= LEVEL_NODE_COUNT) { return; }
+  let nodeIndex = FIRST_NODE + globalId.x;
+  if (atomicLoad(&activeNodes[ACTIVE_OFFSET + nodeIndex]) == 0u) { return; }
+  ${predicate}
+  if (selected) {
+    let leafIndex = nodeIndex - INTERNAL_NODE_COUNT;
+    let objectId = leafIds[LEAF_IDS_OFFSET + leafIndex];
+    if (objectId != ${INVALID_NODE}u) {
```
[the remark this answers is not in the document]
Comment (ibgreen-openai) [reply modules/experimental/src/gpu-primitives/gpu-bvh-query.ts:292]: Addressed in #2877: BVH query now identifies empty leaves from their invalid bounds, preserves 0xffffffff as a valid stable ID, and includes a regression covering the maximum uint32 identity.

File: modules/experimental/src/gpu-primitives/gpu-bvh-query.ts [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+    this.nodeCount = this.bvh.leafCapacity * 2 - 1;
+    this.internalNodeCount = this.bvh.leafCapacity - 1;
+    this.levelCount = Math.log2(this.bvh.leafCapacity) + 1;
+    validateIndexView(this);
+    validatePackedView(this.query, ['float32'], `${this.id} query`);
+    validatePackedUint32View(this.output, `${this.id} output`);
+    validatePackedUint32View(this.count, `${this.id} count`);
+    validatePackedUint32View(this.overflow, `${this.id} overflow`);
+    if (this.outputMask) validatePackedUint32View(this.outputMask, `${this.id} outputMask`);
+    if (this.visitedCount) validatePackedUint32View(this.visitedCount, `${this.id} visitedCount`);
```
[the remark this answers is not in the document]
Comment (ibgreen-openai) [reply modules/experimental/src/gpu-primitives/gpu-bvh-query.ts:101]: Addressed in #2877: the query constructor rejects overlaps between writable results and every live BVH/query input, and rejects overlaps among result views. Regression tests cover leaf/output and count/overflow aliasing.
